conans/cli/api/subapi/graph.py

```diff
@@ -83,11 +84,16 @@ def load_root_test_conanfile(self, path, tested_reference, profile_host, profile
         return root_node
 
     @api_method
-    def load_root_virtual_conanfile(self, ref, profile_host, is_build_require=False,
+    def load_root_virtual_conanfile(self, profile_host, requires=None, tool_requires=None,
                                     require_overrides=None):
+        if not requires and not tool_requires:
+            raise ConanException("Provide requires or tool_requires")
         app = ConanApp(self.conan_api.cache_folder)
-        profile_host.options.scope(ref.name)
-        conanfile = app.loader.load_virtual([ref],  is_build_require=is_build_require,
+        if requires and len(requires) == 1 and not tool_requires:
+            profile_host.options.scope(requires[0].name)
+        if tool_requires and len(tool_requires) == 1 and not requires:
+            profile_host.options.scope(tool_requires[0].name)
```

Comment:
This looks tricky in the API. I get why it is like this, but it makes no sense for the API. Should the profile options be scoped before this call?

Reply:
Yes, I don't love it, I was adding this to not break some existing tests, but could be moved outside the API, yes.

Reply:
Moved to ``scope_options()`` helper.